enhancements/ingress/lb-allowed-source-ranges.md

```diff
@@ -0,0 +1,222 @@
+  - "@knobunc"
+  - "@Miciah"
+  - "@candita"
+  - "@frobware"
+  - "@rfredette"
+  - "@deads2k"
+approvers:
+  - "@frobware"
+  - "@Miciah"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "@deads2k"
+creation-date: 2022-07-08
+last-updated: 2022-07-08
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - "https://issues.redhat.com/browse/NE-555"
+see-also:
+replaces:
+superseded-by:
+---
+
+# LoadBalancer Allowed Source Ranges
+
+## Summary
+
+This enhancement extends the IngressController API to allow the user to specify a 
+list of IP address ranges for an IngressController to which access to the LoadBalancer Service 
+should be restricted when `endpointPublishingStrategy` is `LoadBalancerService`. 
+By default, all source addresses are allowed (which is equivalent to
+specifying "0.0.0.0/0"). Currently, the only way for the customers to limit this is by 
+setting `.Spec.LoadBalancerSourceRanges` field for the LoadBalancer Service 
+in front of the router in cloud environments. This is not desired as we prefer
+the customers use the APIs and do not directly modify the managed resources.
+
+## Motivation
+
+Customer security teams are concerned that the public ELB (created in 
+the installation process by default) is allowing all inbound traffic on 0.0.0.0 
+on ports 80 and 443, so they wish to restrict it. 
+
+Customers have been setting `.Spec.LoadBalancerSourceRanges` of the LoadBalancer 
+Service themselves, something that we did not plan to support but also did not 
+actively reject. We recognize that users consider it now to be an existing feature 
+while we prefer they use APIs and not directly modify managed resources.
+
+The main motivation of this enhancement is to allow the customers through a new API 
+to enable and configure `.Spec.LoadBalancerSourceRanges`.
+
+### User Stories
+
+#### As a cluster admin, when configuring an IngressController, I want to limit access to the load balancer to a specified list of IP address ranges on the cloud provider for this ingress
+
+The cluster admin has the option to modify the IngressController to specify 
+a list of IP address ranges for an IngressController to which access to 
+the load balancer should be restricted. This is available as a Day-2 operation 
+to the cluster admin.
+
+### Goals
+
+Enable the configuration of a range for load balancer via the `IngressControllerSpec`, 
+specifically via the `LoadBalancerStrategy`, with a new parameter `AllowedSourceRanges`, 
+which represents a list of IP address ranges. This is then used by the operator to set
+`.Spec.LoadBalancerSourceRanges` of the LoadBalancer Service in front of the router 
+in cloud environments.
+
+If `.Spec.LoadBalancerSourceRanges` of the LoadBalancer Service is set by the 
```

Comment:
Since `AllowedSourceRanges` is used by the operator to set  `.Spec.LoadBalancerSourceRanges`, in the circumstance that the customer has set a value for `.Spec.LoadBalancerSourceRanges` in 4.11, why can't we translate this to `AllowedSourceRanges`, rather than flagging this cluster as unupgradable? Are there some circumstances where that wouldn't work or would be undesirable?

Reply:
After discussing this with @jerpeter1, we decided to go with "if and only if `AllowedSourceRanges` isn't set but `.Spec.LoadBalancerSourceRanges`  is set, then set `AllowedSourceRanges`  to `.Spec.LoadBalancerSourceRanges`.  Otherwise set `.Spec.LoadBalancerSourceRanges`  to `AllowedSourceRanges`".